sv.c

```diff
@@ -17729,6 +17717,43 @@ Perl_report_uninit(pTHX_ const SV *uninit_sv)
     GCC_DIAG_RESTORE_STMT;
 }
 
+/* This helper function for Perl_sv_setsv_flags is as cold as they come.
+ * We should almost never call it and it will definitely croak when we do.
+ * Therefore it should not matter that it is not close to the main function
+ * or that we make it redo work that the caller already did.
+ * The main aim is to keep Perl_sv_setsv_flags as slim as possible and this
+ * includes keeping the call sites for this function small.
+ */
+void S_sv_setsv_flags_croak(pTHX_ SV * const dsv, SV * const ssv)
+{
+    if (SvIS_FREED(dsv)) {
+        Perl_croak(aTHX_ "panic: attempt to copy value %" SVf
+                   " to a freed scalar %p", SVfARG(ssv), (void *)dsv);
+    }
+    if (SvIS_FREED(ssv)) {
+        Perl_croak(aTHX_ "panic: attempt to copy freed scalar %p to %p",
+                   (void*)ssv, (void*)dsv);
+    }
+
+    if (SvTYPE(ssv) > SVt_PVLV)
+    {
+        const char * const type = sv_reftype(ssv,0);
+        if (PL_op)
```

Comment:
factor out reading global PL_op and very likely fn call inside OP_DESC to a root scope instead 2 copies in 2 branches

Reply:
Done & pushed.